packages/@aws-cdk/aws-cognito/lib/user-pool-email.ts

```diff
@@ -0,0 +1,202 @@
+   */
+  readonly fromEmail: string;
+
+  /**
+   * An optional name that should be used as the sender's name
+   * along with the email.
+   *
+   * @default - no name
+   */
+  readonly fromName?: string;
+
+  /**
+   * The destination to which the receiver of the email should reploy to.
+   *
+   * @default - same as the fromEmail
+   */
+  readonly replyTo?: string;
+
+  /**
+   * The name of a configuration set in Amazon SES that should
+   * be applied to emails sent via Cognito.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cognito-userpool-emailconfiguration.html#cfn-cognito-userpool-emailconfiguration-configurationset
+   *
+   * @default - no configuration set
+   */
+  readonly configurationSetName?: string;
+
+  /**
+   * Required if the UserPool region is different than the SES region.
+   *
+   * If sending emails with a Amazon SES verified email address,
+   * and the region that SES is configured is different than the
+   * region in which the UserPool is deployed, you must specify that
+   * region here.
+   *
+   * Must be 'us-east-1', 'us-west-2', or 'eu-west-1'
+   *
+   * @default - The same region as the Cognito UserPool
+   */
+  readonly sesRegion?: string;
+}
+
+/**
+ * Configuration for Cognito email settings
+ */
+export interface UserPoolEmailConfig {
+  /**
+   * The name of the configuration set in SES.
+   *
+   * @default - none
+   */
+  readonly configurationSet?: string;
+
+  /**
+   * Specifies whether to use Cognito's built in email functionality
+   * or SES.
+   *
+   * @default - Cognito built in email functionality
+   */
+  readonly emailSendingAccount?: string;
+
+  /**
+   * Identifies either the sender's email address or the sender's
+   * name with their email address.
+   *
+   * If emailSendingAccount is DEVELOPER then this cannot be specified.
+   *
+   * @default 'no-reply@verificationemail.com'
+   */
+  readonly from?: string;
+
+  /**
+   * The destination to which the receiver of the email should reply to.
+   *
+   * @default - same as `from`
+   */
+  readonly replyToEmailAddress?: string;
+
+  /**
+   * The ARN of a verified email address in Amazon SES.
+   *
+   * required if emailSendingAccount is DEVELOPER or if
+   * 'from' is provided.
+   *
+   * @default - none
+   */
+  readonly sourceArn?: string;
+}
+
+/**
+ * Configure how Cognito sends emails
+ */
+export abstract class UserPoolEmail {
+  /**
+   * Send email using Cognito
+   */
+  public static withCognito(replyTo?: string): UserPoolEmail {
+    return new CognitoEmail(replyTo);
+  }
+
+  /**
+   * Send email using SES
+   */
+  public static withSES(options: UserPoolSESOptions): UserPoolEmail {
+    return new SESEmail(options);
+  }
+
+
+  /**
+   * Returns the email configuration for a Cognito UserPool
+   * that controls how Cognito will send emails
+   */
+  public abstract bind(scope: Construct): UserPoolEmailConfig;
```

Comment:
I like to use `UserPoolEmailBindOptions` as the parameter to `bind()`. 
This allows for more properties to be added, deprecated, etc. here without breaking changes.

The other option is to mark this API as `@internal` and not export `UserPoolEmailConfig`. Then changes to this API will not be considered breaking.
This option assumes that there aren't going to be many other ways in which customers will want to bind their own implementations here. Seems ok in this case since I don't expect Cognito to support other options than SES.